client/utils/handleSingleAndDoubleClick.js

```diff
@@ -0,0 +1,30 @@
+import { debounce } from 'lodash'
+
+// Due to limitations of ES6 default strict mode on all files, we cannot bind the
+// 'this' object from calling code. Hence, passing this directly as 'self' (ugly).
```

Comment:
I understand you are using `self` to store the debounce. I've red somewhere  that modifying properties of a Component is an anti pattern since two instances of it with the same props could behave differently. But I don't know how we could do otherwise, so it's fine :) Maybe `clickedOnce` could belong to `this.state` ?
One more thing: what about  having `this.clickedOnce = undefined` here, in a wrapper function that would call `onSingleClick()` at the end ? Then all the logic of this `handleSingleAndDoubleClick` function would stay here. What do you think ?